Source/WebKit/WebProcess/Extensions/Cocoa/WebExtensionControllerProxyCocoa.mm

```diff
@@ -110,6 +112,21 @@
     JSObjectSetProperty(context, globalObject, toJSString("chrome").get(), namespaceObject, kJSPropertyAttributeNone, nullptr);
 }
 
+void WebExtensionControllerProxy::addBindingsToWebPageFrameIfNecessary(WebFrame& frame, DOMWrapperWorld& world)
+{
+    auto context = frame.jsContextForWorld(world);
+    auto globalObject = JSContextGetGlobalObject(context);
+
+    auto namespaceObject = JSObjectGetProperty(context, globalObject, toJSString("browser").get(), nullptr);
+    if (namespaceObject && JSValueIsObject(context, namespaceObject))
+        return;
+
+    namespaceObject = toJS(context, WebExtensionAPIWebPageNamespace::create(WebExtensionAPIWebPageNamespace::ForMainWorld::Yes, WebExtensionContentWorldType::WebPage).ptr());
+
+    JSObjectSetProperty(context, globalObject, toJSString("browser").get(), namespaceObject, kJSPropertyAttributeNone, nullptr);
+    JSObjectSetProperty(context, globalObject, toJSString("chrome").get(), namespaceObject, kJSPropertyAttributeNone, nullptr);
```

Comment:
We don't want to add `chrome` this will break a lot of pages and make them think Safari is Chrome.